src/Common/src/TypeSystem/Common/PropertySignature.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Internal.TypeSystem
+{
+    public struct PropertySignature
```

Comment:
If you really insist on kicking this out of the type system, I can do that, but I'll need to make things on EcmaSignatureParser public...